### PR TITLE
Us s4 user security

### PR DESCRIPTION
**Pull request by FlorianHe**
Adding the user and the security to protect the api

### CODE REVIEW

**Review by alexboutemy — APPROVED** [review]

File: src/main/java/com/templateproject/api/entity/Article.java [hunk 1]
```diff
@@ -168,4 +164,16 @@ public List<Paragraph> getListParagraphs() {
     public void setListParagraphs(List<Paragraph> listParagraphs) {
         this.listParagraphs = listParagraphs;
     }
+
+    public void setId(Long id) {
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/entity/Article.java]: no setter for id

File: src/main/java/com/templateproject/api/entity/User.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    private String username;
+
+    @Email
+    @Column(unique = true, nullable = false)
+    private String email;
+
+    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
+    private String password;
+
+    // EAGER, will directly fetch the roles, longer loading time
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/entity/User.java:54]: remove dead code

Comment (Benja-san) [reply src/main/java/com/templateproject/api/entity/User.java:54]: unless it has educational or informative purpose for security explanation you can keep it

File: src/main/java/com/templateproject/api/service/TokenService.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+@Service
+public class TokenService {
+
+    private final JwtEncoder jwtEncoder;
+
+    public TokenService(JwtEncoder jwtEncoder) {
+        this.jwtEncoder = jwtEncoder;
+    }
+
+    //Use Authentication class from security ???
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/service/TokenService.java]: same comment as first one about dead code

File: src/main/java/com/templateproject/api/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            @Lazy AuthenticationManager authManager, TokenService tokenService) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.commentRepository = commentRepository;
+        this.passwordEncoder = passwordEncoder;
+        this.authManager = authManager;
+        this.tokenService = tokenService;
+    }
+
+    // Security method to implement
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/service/UserService.java]: See comment about dead code

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Review by alexboutemy — APPROVED** [review]

**Pull request merged**